From 879abfa1d86f820326bd8edab5e7d5c8991cc9f0 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sun, 3 Feb 2013 21:36:30 +0100
Subject: [PATCH 10/17] Revert "A new try to get the parallelism straight."

This reverts commit 369ac3b365091ba1a26fb266a94d97adb3719cbc
and obviously this is not a solution either.
---
 python/pakfire/system.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/python/pakfire/system.py b/python/pakfire/system.py
index 6f67d31..a9154de 100644
--- a/python/pakfire/system.py
+++ b/python/pakfire/system.py
@@ -156,10 +156,15 @@ class System(object):
 			Calculates how many processes should be run
 			simulatneously when compiling.
 		"""
+		# Check how many processes would fit into the
+		# memory when each process takes up to 500MB.
+		multiplicator = self.memory / (500 * 1024 * 1024)
+		multiplicator = round(multiplicator)
+
 		# Count the number of online CPU cores.
 		cpucount = os.sysconf("SC_NPROCESSORS_CONF")
 
-		return cpucount * 2
+		return min(multiplicator, cpucount * 2)
 
 
 # Create an instance of this class to only keep it once in memory.
-- 
1.8.1.2

